### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by Alex-Izquierdo**
Fixes https://issues.redhat.com/browse/AAP-6908 and https://github.com/ansible/ansible-rulebook/issues/226

Includes: 
- Reorganize content from "rules" section into "conditions" and "actions" and remove some duplicated content
- Set "text" as the default syntax (by default is python which is a bit ugly when the majority of the code blocks are other formats)
- Remove the dollar sign in shell code blocks to allow copy/paste
- Add "work in progress" in missing or incomplete sections
- Fix links references for https://ansible-rulebook.readthedocs.io/
- Fix some missing code blocks

How to test: 

```sh
cd docs
make html
cd _build/html
python -m http.server 3000
xdg-open http://localhost:3000/
```

### CODE REVIEW

**Comment by Alex-Izquierdo** [comment]
@benthomasson The docs website must be updated if this PR is merged. 

Should we add a GH action to update automatically https://ansible-rulebook.readthedocs.io/ ?

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -77,35 +77,37 @@ Ready to contribute? Here's how to set up `ansible_rulebook` for local developme
...
 
 5. Create a branch for local development::
 
-    $ git checkout -b name-of-your-bugfix-or-feature
+    git checkout -b name-of-your-bugfix-or-feature
 
    Now you can make your changes locally.
 
 6. When you're done making changes, check that your changes pass flake8 and the
    tests, including testing other Python versions with tox::
```
Comment (bzwei) [inline CONTRIBUTING.rst:98]: same. apply `.. code-block:: console`

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -181,7 +183,7 @@ Tips
 
 To run a subset of tests::
```
Comment (bzwei) [inline CONTRIBUTING.rst:184]: same, apply `.. code-block:: console`

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -138,31 +140,31 @@ framework to automatically setup hooks for you.
 
      .. code-block:: console
 
-         (venv) $ pip install pre-commit
+         (venv) pip install pre-commit
```
[the remark this answers is not in the document]
Comment (Alex-Izquierdo) [reply CONTRIBUTING.rst]: good catch, thanks!

File: docs/conditions.rst [hunk 4]
```diff
@@ -36,3 +253,63 @@ Conditions support the following operators:
...
+      name: rule1
+      condition: event.msg is defined
+      action:
+        retract_fact:
+          fact:
+            msg: "{{event.msg}}"
+
+| **Q:** How do I check if an attribute in an object referred in a condition does not exist?
+
+| **Ans:** Use the is not defined
```
Comment (ttuffin) [inline docs/conditions.rst:306]:
```suggestion
| **Ans:** Use the "is not defined" logical operator
```

File: docs/conditions.rst [hunk 5]
```diff
@@ -36,3 +253,63 @@ Conditions support the following operators:
...
+   In the above case if any of the event.payload.eventType is undefined the condition is
+   ignored and doesn't match anything.
+
+| **Q:** When a rule book has multiple rule sets and one of them shuts down are all rule sets terminated?
+
+| **Ans:** Yes, so care should be taken if there are any playbooks running in the other rule sets
+
+| **Q:** How do I check if an attribute in an object referred in a condition exists?
+
+| **Ans:** Use the is defined
```
[the remark this answers is not in the document]
Comment (Alex-Izquierdo) [reply docs/conditions.rst:293]: To be strict, it's an identity operator. Logical operators would "and" and "or"

File: docs/conditions.rst (hunk 4 above)
Comment (Alex-Izquierdo) [reply docs/conditions.rst:306]: To be strict, it's an identity operator. Logical operators would "and" and "or"

Comment (ttuffin) [reply docs/conditions.rst:306]: Fixed

**Comment by benthomasson** [comment]
@Alex-Izquierdo please rebase

**Comment by Alex-Izquierdo** [comment]
I will remade this in a new PR.

**Pull request closed without merging**